crypto/fipsmodule/cpucap/cpu_aarch64_linux.c

```diff
@@ -29,29 +29,45 @@
 #include <string.h>
 
 #include <openssl/arm_arch.h>
+#include <openssl/err.h>
 
+OPENSSL_DECLARE_ERROR_REASON(SYS, CPU_R_UNSUPPORTED_HW_CAPABILITY);
 
 extern uint32_t OPENSSL_armcap_P;
 
-// handle_cpu_env applies the value from |in| to the CPUID values in |out[0]|
-// and |out[1]|. See the comment in |OPENSSL_cpuid_setup| about this.
+// handle_cpu_env applies the value from |in| to the CPUID values in |out[0]|.
+// See the comment in |OPENSSL_cpuid_setup| about this.
 static void handle_cpu_env(uint32_t *out, const char *in) {
   const int invert = in[0] == '~';
   const int or = in[0] == '|';
   const int skip_first_byte = invert || or;
   const int hex = in[skip_first_byte] == '0' && in[skip_first_byte+1] == 'x';
+  uint32_t armcap = out[0];
 
   int sscanf_result;
   uint32_t v;
   if (hex) {
-    sscanf_result = sscanf(in + invert + 2, "%" PRIx32, &v);
+    sscanf_result = sscanf(in + skip_first_byte + 2, "%" PRIx32, &v);
   } else {
-    sscanf_result = sscanf(in + invert, "%" PRIu32, &v);
+    sscanf_result = sscanf(in + skip_first_byte, "%" PRIu32, &v);
   }
 
   if (!sscanf_result) {
     return;
   }
+  // If getauxval() returned a non-zero hwcap in `armcap` (out)
```

Comment:
Can we add something like this in the beginning of the comment: `Detect if the user is trying to use the environment variable to set a capability that is _not_ available on the CPU.`?

Reply:
Done